src/sinks/util/mod.rs

```diff
@@ -132,6 +435,66 @@ pub fn encode_log(mut event: Event, encoding: &EncodingConfig<Encoding>) -> Opti
                 .unwrap_or_default();
             Ok(bytes)
         }
+        Encoding::Syslog(config) => {
+            let mut buf = String::from("<");
+            let pri = get_num_facility(&config.facility, &log) * 8 + get_num_severity(&config.severity, &log);
+            buf.push_str(&pri.to_string());
```

Comment:
doing `push_str` so many times does not seem very efficient. We can do some pre-allocation, or use `concat_string` macro.
https://github.com/hoodie/concatenation_benchmarks-rs#concat_string_macro

Reply:
That's an interesting find, there's a benchmark for everything ;)
Agreed, this is naive and I hoped push_str() reallocates exponentially.